proto/kava/cdp/v1beta1/genesis.proto

```diff
@@ -0,0 +1,146 @@
+}
+
+// Params defines the parameters for the cdp module.
+message Params {
+  option (gogoproto.goproto_stringer) = false;
+
+  repeated CollateralParam collateral_params = 1
+      [(gogoproto.castrepeated) = "CollateralParams", (gogoproto.nullable) = false];
+  DebtParam debt_param = 2 [(gogoproto.nullable) = false];
+
+  cosmos.base.v1beta1.Coin global_debt_limit         = 3 [(gogoproto.nullable) = false];
+  string                   surplus_auction_threshold = 4 [
+    (cosmos_proto.scalar)  = "cosmos.Int",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable)   = false
+  ];
+  string surplus_auction_lot = 5 [
+    (cosmos_proto.scalar)  = "cosmos.Int",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable)   = false
+  ];
+  string debt_auction_threshold = 6 [
+    (cosmos_proto.scalar)  = "cosmos.Int",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable)   = false
+  ];
+  string debt_auction_lot = 7 [
+    (cosmos_proto.scalar)  = "cosmos.Int",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable)   = false
+  ];
+  bool circuit_breaker = 8;
+}
+
+// DebtParam defines governance params for debt assets
+message DebtParam {
+  option (gogoproto.goproto_stringer) = false;
+
+  string denom             = 1;
+  string reference_asset   = 2;
+  string conversion_factor = 3 [
+    (cosmos_proto.scalar)  = "cosmos.Int",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable)   = false
+  ];
+  string debt_floor = 4 [
+    (cosmos_proto.scalar)  = "cosmos.Int",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable)   = false
+  ];
+}
+
+// CollateralParam defines governance parameters for each collateral type within the cdp module
+message CollateralParam {
+  option (gogoproto.goproto_stringer) = false;
+
+  string denom             = 1;
+  string type              = 2;
+  string liquidation_ratio = 3 [
+    (cosmos_proto.scalar)  = "cosmos.Dec",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable)   = false
+  ];
+  cosmos.base.v1beta1.Coin debt_limit    = 4 [(gogoproto.nullable) = false];
+  string                   stability_fee = 5 [
+    (cosmos_proto.scalar)  = "cosmos.Dec",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable)   = false
+  ];
+  string auction_size = 6 [
+    (cosmos_proto.scalar)  = "cosmos.Int",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable)   = false
+  ];
+  string liquidation_penalty = 7 [
+    (cosmos_proto.scalar)  = "cosmos.Dec",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable)   = false
+  ];
+  // No single byte type, use a uint32
+  uint32 prefix                   = 8;
```

Comment:
There isn't a single `byte` type for protobuf so not sure which one would be preferred:

`uint32` and casting between byte/uint32 in places like keeper vs `bytes` and just using the first item.  Leaning towards just `uint32` as it represents a single item vs possible multiple ones with `bytes` type which seems less clear and could maybe be prone to indexing errs.

Reply:
👍  I'll use the collateral type, not too clear why there's the separate prefix originally -- is there a need to keep store key/values short?

Reply:
No strong need to keep them short. But they do need to allow iteration. There's a few keeper methods that iterator over cdps with the same collateral-type.

I wrote up some notes on different encoding options here: https://kava-labs.atlassian.net/wiki/spaces/ENG/pages/1030488065/Store+Key+Encoding.